### ISSUE TITLE
issues with creating backups.

### ISSUE THREAD

**Issue by mglee72** [cptactionhank/docker-atlassian-confluence#7]
Hi, I have an issue with creating backups in admin page.
Fantastic work, btw.

I am using the latest version (5.7).
I am trying to make a backup to /var/local/atlassian/confluence/backups/ by selecting a checkbox which enables xml export backups to be created in backups/ not only in temp/ directory.

I have absolutely no problem with generating backups in temp directory, however, whenever I try it on backups/ directory, it fails with following exception:

Java.io.IOException: No such file or directory
	at java.io.UnixFileSystem.createFileExclusively(Native Method)
	at java.io.File.createNewFile(File.java:1006)
	at com.atlassian.core.util.FileUtils.copyFile(FileUtils.java:461)
	at com.atlassian.core.util.FileUtils.copyFile(FileUtils.java:397)
	at com.atlassian.confluence.importexport.actions.BackupAction.execute(BackupAction.java:77)
	at com.opensymphony.xwork.DefaultActionInvocation.invoke(DefaultActionInvocation.java:168)
        .........

backups directory does exist and I have also tried granting write permission but it did not help at all.

One possible solution might be modifying confluence.cfg.xml to allow custom path for backups to try different directories other than backups/. However, this would mean that modifications should me made to the original image, since to my knowledge, I cannot use any editors (nano, vi) inside running container by attaching with bash.

I have not had a chance to observe if daily backups (2:00AM) have also failed or not. I will get a result probably tomorrow morning. Any ideas?

**Comment by cptactionhank** [comment]
Does it still fail? 

I will do my best to look into this and see if I'm experiencing the same thing on my running instance, right now i'm using my spare time on making automated tests of my Docker JIRA repository and some improvements there.

At some point I changed the default running user of the image from `nobody` to the `daemon` user instead, perhaps this is what's causing trouble if using the data container pattern or host mounting some directories?

If you figure it's a problem in the contents of the Dockerfile I do accept PR's.

**Comment by mglee72** [comment]
Start Image -> new or example confluence with embedded db can successfully create backups.
Start Image -> restore from existing backup file fails to create backups in backups/ directory.

I have identified the source of the problem. It seems that before migration, the original image had it's backup directory set to somewhere else than /var/local/atlassian/confluence/backups/. In my case, it was set to /usr/local/atlassian/confluence-data/backups/. 

As I ran the image and restored from existing backup, the admin page in the browser pointed to /var/local/atlassian/confluence/backups/ whereas the "internal" backup directory was set to the one restored from the backup image (/usr/local/atlassian/confluence-data/backups/).

I have no extensive knowledge of the internals of Confluence, but to me, it seems like this problem is Confluence-specific. I have seen some other posts on the web stating the exact same problems. 

I will try changing the backup directory (I've modified the confluence configuration xml file to allow custom locations for backups), but I am currently having issues with restarting confluence. I will post a follow up afterwards.

Thank you.

**Comment by mattstein** [comment]
FWIW, I experienced the exact same issue with a fresh install of Confluence 6.0.3. I restored data from a previous 5.x version, and yet my scheduled backups were silently failing. When attempting a manual backup, Confluence gave me an error about the directory not existing. I set `admin.ui.allow.daily.backup.custom.location` to `true`, and the default value of `/var/atlassian/confluence/backups` on the Backup Administration settings page was set (without my intervention) to another directory that didn't exist. I entered the manual value of `/var/atlassian/confluence/backups` again since the directory exists, and a manual backup suddenly worked just fine. Even though the default value and "custom location" were the same directory, Confluence only seemed to be happy when I _treated_ that reference as a custom location.
